reporting-anything/src/download_helpers.js

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import puppeteer from 'puppeteer-core';
+import createDOMPurify from 'dompurify';
+import fs from 'fs';
+import { JSDOM } from 'jsdom';
+import { CHROMIUM_PATH, REPORT_TYPE_URLS, FORMAT, REPORT_TYPE, SELECTOR } from './constants.js';
+
+
+export async function downloadVisualReport(url, type, object_id, format, width, height) {
+  const window = new JSDOM('').window;
+  const DOMPurify = createDOMPurify(window);
+
+  try {
+    const browser = await puppeteer.launch({
+      headless: true,
+      /**
+       * TODO: temp fix to disable sandbox when launching chromium on Linux instance
+       * https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md#setting-up-chrome-linux-sandbox
+       */
+      args: [
+        '--no-sandbox',
+        '--disable-setuid-sandbox',
+        '--disable-gpu',
+        '--no-zygote',
+        '--single-process',
+        '--font-render-hinting=none',
+      ],
+      executablePath: CHROMIUM_PATH,
+      ignoreHTTPSErrors: true,
+      env: {
+        TZ: 'UTC', // leave as UTC for now
+      },
+    });
+
+    const page = await browser.newPage();
+    page.setDefaultNavigationTimeout(0);
+    page.setDefaultTimeout(100000);
+
+    let queryUrl = '';
+    if (url != undefined) {
+      queryUrl = url;
+    }
+    else {
+      queryUrl = 'http://localhost:5601/app/';
```

Comment:
why do we want to support this? the use case seems limited (doesn't work if host, port, or basePath is different from defaults)